### ISSUE TITLE
Park Specific Confirmation Messaging

### ISSUE THREAD

**Issue by JLWade** [bcgov/parks-reso-public#345]
#### Description:
NEEDS TO BE BROKEN DOWN AND SPLIT
[User Story]

[Description of the ticket]

#### Acceptance Criteria:

Given
When
Then 

Given
When
Then

#### Development Checklist:

- [ ] ...
- [ ] ...
- [ ] ...

#### Dependencies

- Blocked by 
- Blocking

**Wireframe(s):**

**Workflows:**

**Definition of Ready** (Note: If any of these points are not applicable, mark N/A)

- [ ] User story is included
- [ ] User role and type are identified
- [ ] Acceptance criteria are included
- [ ] Wireframes are included
- [ ] Design / Solution is accepted by Product Owner
- [ ] Dependencies are identified (technical, business, regulatory/policy)
- [ ] Story has been estimated (under 13 pts)

**Definition of Done** 
- In progress:
  - [ ] Acceptance criteria are tested (Functionality meets the acceptance criteria defined in the ticket)
  - [ ] UI meets accessibility requirements
  - [ ] Unit tests are written
  - [ ] Work is traceable in GitHub
  - [ ] PR linked to ticket number
  - [ ] If needed/required - Dev adds flag/label to highlight any migration steps necessary prior to PROD deployment
- Code review:
  - [ ] Code is peer reviewed and has passed CI/CD tests
- QA:
  - [ ] Acceptance criteria are tested (Functionality meets the acceptance criteria defined in the ticket)
  - [ ] Code is potentially shippable to the production environment
  - [ ] Functional features have been tested and passed by QA
  - [ ] UI components tested by designer
  - [ ] Code is deployed to PROD when moved to 'done' column (unless requested otherwise by PO)
- PO Review:
  - [ ] Acceptance criteria are tested (Functionality meets the acceptance criteria defined in the ticket)
  - [ ] Reviewed and approved by Product Owner

#### Notes:
- 
-

**Comment by Dianadec** [comment]
@meyerdarcie are we able to populate the AC in this ticket and break it out into separate pieces (as needed) based on the MS Teams thread linked? This work will depend on designs in that other ticket of course, but if we already know what needs to be designed, we could try to refine this ticket further already.

**Comment by Dianadec** [comment]
Blocked by icons license.

**Comment by Dianadec** [comment]
@ndelraye fyi we now have the FA License so I will unblock this. However, let me know if the wireframes need to be updated according to Marshall's last update. cc @meyerdarcie

**Comment by Dianadec** [comment]
@ndelraye also, is the source of truth for the text in this one going to be the word doc? Or this ticket?

**Comment by ndelraye** [comment]
@meyerdarcie @Dianadec - wireframes are up to date as per last conversation with Marshall and this ticket, the word document and wireframes all have the current and final text represented

**Comment by cameronpettit** [comment]
@Dianadec @meyerdarcie I just want to doubly confirm that the plan with this ticket is to make the changes included ONLY if the user is booking for Joffre Lakes Provincial Park. If they are booking for any other park, the current designs will remain as they currently are. Is that correct?

**Comment by meyerdarcie** [comment]
Yes, that's correct Cam - these changes are specific to Joffre ONLY. 

Coming up, there is an additional ticket for changes to Garibaldi and in future, Golden Ears.

**Comment by cameronpettit** [comment]
Where to put this FontAwesome helper doc?

# Working with FontAwesome 
If you're feeling lucky, check out the official [FontAwesome instructions](https://docs.fontawesome.com/apis/javascript/get-started/).

### Package Token
A package token is needed to load the appropriate FontAwesome package.
1. Obtain a package token for FontAwesome.
2. Store the package key in your `.bashrc` file:
	```
	export FONTAWESOME_PACKAGE_TOKEN='XXXXXXXXXXXXXXXX'
	```
3. Reload `.bashrc` to load the environment variable. 

### Create .npmrc file
In the same directory as your `package.json`, create a file `.npmrc` that looks like the following:
```
@awesome.me:registry=https://npm.fontawesome.com/
@fortawesome:registry=https://npm.fontawesome.com/
//npm.fontawesome.com/:_authToken=${FONTAWESOME_PACKAGE_TOKEN}
```

### Load correct FontAwesome packages
Look [here](https://docs.fontawesome.com/web/setup/packages/) for a list of available FontAwesome packages. Note that the `npm` collection is called `@fortawesome` with an 'r' because it's quirky!!!
1. Install the appropriate FontAwesome package.
For example, for 'classic, regular' icons, the installation would be:
```
yarn add @fortawesome/pro-regular-svg-icons
```
Your [`FONTAWESOME_PACKAGE_TOKEN`](#package-token) should be loaded into your environment already.

2. If using Angular, load the Angular helper package.
```
yarn add @fortawesome/angular-fontawesome
```
The Angular package shouldn't need a package token.

### Create Components with Icons
1. For Angular `v^17.0.0`, the FontAwesome module should be loaded directly into the component. For earlier versions of Angular, you can load the module into the component's parent `module` to make it available.

```
import { FontAwesomeModule } from  '@fortawesome/angular-fontawesome';
```
2. In the component, load the icon.
```
import { faMagnifyingGlass} from '@fortawesome/pro-regular-svg-icons'
```
Note that there is no library that links the design of an icon to its JS import name, _ie_ `faMagnifyingGlass`.  You can look up the icons on the [FontAwesome library](https://fontawesome.com/search) to get the original filename (`magnifying-glass`), convert it to camel case and add `fa` to the front.

* `house` becomes `faHouse`
* `alarm-clock` becomes `faAlarmClock`
* `arrow-down-to-line` becomes `faArrowDownToLine`

3. Within the component declaration, declare a public variable for the icon so it can be accessed by the component's `html`.
```
public faMagnifyingGlass = faMagnifyingGlass
```
4. Reference the icon in the component's `html` file.
```
<fa-icon [icon]="faMagnifyingGlass"></fa-icon>
```

**Comment by Dianadec** [comment]
@cameronpettit How about similar to how we documented info about bootstrap? I see this --> https://github.com/digitalspace/bcparks-bootstrap-theme that I found by going here --> https://bcgov.github.io/bcparks/design-guides

**Comment by cameronpettit** [comment]
@manuji this has received an update as of 7/26/2024 @2:30pm so you may need to deploy. 

Leaving the old instructions above in case we ever need to install via packages again. The new way of including icons is here:

https://docs.fontawesome.com/web/setup/host-yourself/webfonts

Note that exact folder hierarchy is important for this to work.

**Comment by manuji** [comment]
Tested on TEST: **Pending**

Given below are the UI issues noticed;
1. Missing the horizontal white line separating the BCParks logo and the footer links. 
2.  The space between the vertical line dividing the two left and right sections and the right side content is not as per the figma. 
3. All the fontAwesome icons are not exactly 36px, I think that's fine as the ticket AC states that should not be more than 36px. @ndelraye please correct me if I am wrong. 

FYI @cameronpettit

**Comment by cameronpettit** [comment]
@manuji I think a lot of your points above are valid but don't directly block this ticket from progressing - and I say that because we've been trying to get this ticket out for some time now but have run into complications with things like ensuring the facility messaging is truly 'facility specific', FontAwesome, etc. In a perfect world I think the site should reflect the Figma directly but pragmatically its not always possible, and I've seen this issue happen so much now I'm going to make an example of this ticket:

First, and most importantly, there have been major sweeping changes to our design standards lately, not just with FontAwesome but with our colours, font sizes, etc. Our bcparks bootstrap package (has nothing to do with icons) dictates the colour scheme of the website. Recently I believe it was found to be out-of-spec with the BC Parks Design Guideline standards, so the package colours were updated to match the standards. Our products look understandably weird now because some of the shades of blue have been recently updated (@danieltruong correct me if I am wrong). I think the Figma designs are also out-of-spec and have not been updated with matching colours to the bootstrap package (@meredithom correct me if I am wrong). 

Secondly, the way our bootstrap framework is set up (again, nothing to do with icons), we never set any sizes or spacing distances with pixels. It's poor responsive design practice when we want the relative sizes of elements on the screen to adapt to the size of the screen itself. Look at the [bootstrap spacing](https://getbootstrap.com/docs/5.2/utilities/spacing/) examples to get an idea of how we implement our spacing. 
Ultimately, the spacing and sizing of elements is rarely fixed in a bootstrap framework, rather, they are adaptive and a function of the dimensions of the screen itself, so to request/enforce any kind of pixel-based sizing restrictions is pointless. The devs have always just taken the closest bootstrap equivalent to the requested pixel size in the past, but they will never be identical.

@meyerdarcie I'm making a big deal about this because it seems our bootstrap settings in Figma and in our products have been out of sync for a while which has very often led to tickets like this getting held up in a loop of minor UI discrepancies. I'm not really sure what the state of synchronizing the two is either. My bottom line is: if we are going to hold up tickets for UI discrepancies like this, make it easier for developers by providing designs they can implement without having to count pixels or get the eyedropper tool out to check colours. Let me know if you need the above issues corrected or whether they can be pushed into a different UI bug ticket.

I would love @pavelcreates' take on this!

**Comment by Christopher-walsh22** [comment]
I am also following this closely, as this discussion will influence the implementation of ticket #401.

**Comment by meyerdarcie** [comment]
@cameronpettit - thanks for the descriptive explanation. I agree with not blocking this ticket further with the UI improvements that are required, and approve this (and other DUP facility specific messaging work) to move forward despite these design/UI challenges. 

Next steps will be to determine a path forward for the UI improvements in a new ticket, and more broadly, address inconsistencies with our design guidelines/DUP figma, and overall need to revisit DUP front end.  

@pavelcreates - feel free to provide any further insights to Cam's comments above. 

CC: @Dianadec @MarshallHenson @manuji

**Comment by Christopher-walsh22** [comment]
I had to refactor some of this code to work with the garibaldi messaging. Many of the issues @manuji were addressed in doing so. I have not made of the larger changes (Text font size, Line in footer) that would effect the other parts of the app.  Moving to code review now.

**Comment by manuji** [comment]
Acknowledged the improvements. 

Passing to PO review.

**Comment by meyerdarcie** [comment]
UI improvements complete, working as intended, moving to done.
